emma-language/src/main/scala/org/emmalanguage/ast/Trees.scala

```diff
@@ -161,11 +252,58 @@ trait Trees { this: AST =>
           case Lambda(fun, _, _) => fun
         }: _*)(tree)
 
+      /** Renames `tree` with the transitive closure of `aliases`. */
+      def renameTransitive(aliases: (u.Symbol, u.Symbol)*): u.Tree => u.Tree =
+        if (aliases.isEmpty) identity else tree => {
+          val trans = mutable.Map(aliases: _*).withDefault(identity)
+          for (Def(sym) <- tree) {
+            val alias = trans(sym)
+            val owner = trans(alias.owner)
+            if (alias.isMethod) {
```

Comment:
I also would not change this `trans` as the closure does not seem to be transitive, maybe refer to this as `Def`-closure in the comment and name it `defcl`?